src/cdk-experimental/scrolling/virtual-scroll.md

```diff
@@ -0,0 +1,143 @@
+  </div>
+</cdk-virtual-scroll-viewport>
+```
+
+A `trackBy` function can be specified and works the same as the `*ngFor` `trackBy`. The `index`
+passed to the tracking function will be the index in the data source, not the index in the rendered
+portion.
+
+#### View recycling
+In order to improve performance `*cdkVirtualFor` saves the views it creates in a cache when they are
+no longer needed. This way the next time a new view is needed one can be recycled from the cache
+rather than being created from scratch. The size of this cache can be adjusted using the
+`templateCacheSize` input. Caching can be disabled by setting it to `0`.
+
+```html
+<cdk-virtual-scroll-viewport itemSize="50">
+  <div *cdkVirtualFor="let item of items; templateCacheSize: 0">{{item}}</div>
+</cdk-virtual-scroll-viewport>
+```
+
+#### Specifying items with an Observable or DataSource
+`*cdkVirtualFor` is set up to accept data from an `Array`, `Observable`, or `DataSource`. The
+`DataSource` for the virtual scroll is the same one used by the table and tree components. A
+`DataSource` is simply an abstract class that has two methods: `connect` and `disconnect`. The
+`connect` method will be called by the virtual scroll viewport to receive a stream that emits the
+data array that should be rendered. The viewport will call `disconnect` when the viewport is
+destroyed, which may be the right time to clean up any subscriptions that were registered during the
+connect process.
+
+### Scrolling over fixed size items
+If you're scrolling over a list of items that are all the same fixed size, you can use the
+`FixedSizeVirtualScrollStrategy`. This can be easily added to your viewport using the `itemSize`
+directive.
+
+```html
+<cdk-virtual-scroll-viewport itemSize="50">
+  ...
+</cdk-virtual-scroll-viewport>
+```
+
+The fixed size strategy also supports setting the buffer size, i.e. the number of items rendered
+beyond the edge of the viewport. This can be adjusted by setting the `bufferSize` input.
+
+```html
+<cdk-virtual-scroll-viewport itemSize="50" bufferSize="1">
+  ...
+</cdk-virtual-scroll-viewport>
+```
+
+**Note: The fixed size strategy will likely be changed to allow specifying a separate
+`minBufferPx` and `addBufferPx` like the autosize strategy**
+
+The fixed size viewport strategy is less flexible than the autosize strategy because it requires all
+items to be the same size, but the advantage of this constraint is that it allows for better
+performance, since items do not need to be measured as they are rendered. 
+
+### Scrolling over items with different sizes
+If you're scrolling over a list of items with different sizes, you can use the
+`AutoSizeVirtualScrollStrategy`. This can be added to your viewport by using the `autosize`
+directive.
+
+```html
+<cdk-virtual-scroll-viewport autosize>
+  ...
+</cdk-virtual-scroll-viewport>
+```
+
+The `autosize` strategy allows the buffer to be configured through two inputs `minBufferPx` and
```

Comment:
I would break up the [paragraph structure a bit here:
```markdown
The `autosize` strategy is configured through two inputs: `minBufferPx` and `addBufferPx`.

**`minBufferPx`** determines the minimum space outside virtual scrolling viewport that will be filled
with content. Increasing this will increase the amount of content a user will see before more content
must be rendered. However, too large a value will cause more content to be rendered than is necessary.

**`addBufferPx`** determines the amount of content that will be added incrementally as
the viewport is scrolled. This should be greater than the size of `minBufferPx` so that one
"render" is needed at a time.
```